src/dda/tools/git.py

```diff
@@ -234,28 +243,39 @@ def get_changes(
         # Filter out any empty patches
         patches = [patch.strip() for patch in patches if patch.strip()]
 
-        return ChangeSet.generate_from_diff_output(patches)
+        return ChangeSet.from_patches(patches)
 
     def _get_working_tree_patch(self) -> str:
-        from os import environ
+        from dda.utils.fs import temp_directory
 
-        from dda.utils.fs import temp_file
-
-        with temp_file(suffix=".git_index") as temp_index_path:
+        with temp_directory() as temp_dir:
```

Comment:
I'm not precisely certain how this was passing on Windows CI but locally the usage of an already opened file fails:

```
fatal: unable to write new index file

Command failed with exit code 128: ['C:\\Program Files\\Git\\cmd\\git.EXE', 'read-tree', 'HEAD']
```